test/boost/chunked_vector_test.cc

```diff
@@ -241,3 +241,52 @@ BOOST_AUTO_TEST_CASE(test_amoritzed_reserve) {
     amortized_reserve(v, 1);
     BOOST_REQUIRE_EQUAL(v.capacity(), 8);
 }
+
+struct push_back_item {
+    std::unique_ptr<int> p;
+    push_back_item() = default;
+    push_back_item(int v) : p(std::make_unique<int>(v)) {}
+    // Note: the copy constructor adds 1 to the copied-from value
+    // so it can be checked by test, in constrast to the move constructor.
+    push_back_item(const push_back_item& x) : push_back_item(x.value() + 1) {}
+    push_back_item(push_back_item&& x) noexcept : p(std::exchange(x.p, nullptr)) {}
+
+    int value() const noexcept { return *p; }
```

Comment:
nit, s/value()/seq_id_of_this_copy()/ this should partially address Nadav's confusion.